hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1439,6 +1440,238 @@ private class Emit(
         }
         emitter.emit(outputPType)
 
+      case x@NDArrayQR(nd, mode) =>
+        val ndt = emit(nd)
+        val ndAddress = mb.newField[Long]
+        val ndPType = nd.pType.asInstanceOf[PNDArray]
+
+        val shapeAddress = ndPType.shape.load(region, ndAddress)
+        val shapeTuple = new CodePTuple(ndPType.shape.pType, region, shapeAddress)
+        val shapeArray = (0 until ndPType.shape.pType.nFields).map(shapeTuple[Long](_)).toArray
+
+        val sizeQueryAddress = mb.newLocal[Long]
+
+        val M = shapeArray(0)
+        val N = shapeArray(1)
+        val K = (M < N).mux(M, N)
+        val LDA = M // Possible stride tricks could change this in the future.
+        val LWORK = Region.loadDouble(sizeQueryAddress).toI
```

Comment:
It's a bit confusing that LWORK is just the load of a value not also called LWORK. I prefer consistency in either direction. I guess I prefer universally using LWORK. Working on this code necessitates understanding LAPACK, so I think you should embrace their terminology when it improves understanding.

Reply:
Ok, renamed `sizeQueryAddress` to `LWORKAddress`